source/jsonParsing.ml

```diff
@@ -64,3 +64,201 @@ let optional_list_member ~f name json =
 let string_list_member = list_member ~f:to_string
 
 let path_list_member = list_member ~f:to_path
+
+module Location = struct
+  type t = {
+    start_line: int;
+    end_line: int;
+    start_column: int;
+    end_column: int;
+  }
+  [@@deriving show, eq]
+
+  let from_decoded_range range =
+    let (start_line, end_line), (start_column, end_column) = range in
+    { start_line; end_line; start_column; end_column }
+end
```

Comment:
`Ast.Location.t` seems to have a slightly better design that avoids repetition:
```
type position = { line: int; column: int }
type t = { start: position; end: position }
```

Reply:
Thanks for that, much better